samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java

```diff
@@ -79,11 +79,14 @@ public void testLifecycleListenerAfterFailure() {
   public void testShutdownHookInvokesShutdownHookCallback() {
     doNothing().when(mockShutdownHookCallback).run();
 
-    // setup shutdown hook
+    // call beforeStart to setup shutdownHook
```

Comment:
I think the timeout is actually set to 1ms in the setup, so this still might trigger the timeout. Could you set the timeout to something bigger? Then at least some could manually see the log to verify that the timeout did not get hit.

Reply:
Ah your right ! I moved the `afterStop()` into the callback so we can verify the invocation.